plugins/clamd.js

```diff
@@ -315,6 +327,18 @@ exports.hook_data_post = function (next, connection) {
     try_next_host();
 }
 
+exports.wants_skip = function (connection) {
+    const plugin = this;
+
+    if (!plugin.cfg.check.authenticated && connection.notes.auth_user) return true;
+
+    if (connection.remote.is_local) return !plugin.cfg.check.local_ip;
```

Comment:
Same here, lets stay consistent with the code surrounding this line. The biggest reason is future maintainability. Someone else might add another condition after this and be surprised when it sometimes doesn't run.

```suggestion
    if (plugin.cfg.check.local_ip && connection.remote.is_local) return true;
```

I'm half tempted to say, move the `txn.results.add(plugin, {skip: 'criteria match'});` code into this function, so that the skip message can be a more precise reason (authed, localhost, private).